lib/puppet/runtime.rb

```diff
@@ -0,0 +1,25 @@
+require 'puppet/http'
+
+class Puppet::Runtime
+  include Singleton
+
+  def initialize
+    @runtime_services = {}
+    self['http'] = proc { Puppet::HTTP::Client.new }
```

Comment:
The `http` service should be registered from outside this class, so that we can configure the client